### ISSUE TITLE
http: panic when loading template for "Create Cluster" workflow

### ISSUE THREAD

**Issue by kingdonb** [weaveworks/weave-gitops-enterprise#2871]
tl;dr: The error:

```
http: panic serving 10.32.0.8:34990: runtime error: index out of range [0] with length 0 goroutine
```

on
```
weave-gitops-enterprise	flux-system	1       	2023-05-15 02:58:03.951100271 +0000 UTC	deployed	mccp-0.23.0                  	1.16.0
```

with definition here:

https://github.com/kingdonb/bootstrap-repo/tree/main/apps/howard-wge/weave-gitops

For additional context, it was just recently that I installed `ww-gitops` the open source Weave GitOps portal to this same cluster, when it hadn't been installed before. I think some workarounds were necessary to make them both install concurrently. That may or may not have anything to do with the error that popped up.

That definition is also in here:

https://github.com/kingdonb/bootstrap-repo/blob/main/apps/howard-wge/weave-gitops/helm-artifacts.yaml
**Kubernetes release:**

```
$ k version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.1", GitCommit:"4c9411232e10168d7b050c49a1b59f6df9d7ea4b", GitTreeState:"clean", BuildDate:"2023-04-14T13:14:41Z", GoVersion:"go1.20.3", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.1+k3s1", GitCommit:"f10af367c3c96863c081ada4018e94e085c9404d", GitTreeState:"clean", BuildDate:"2023-01-26T00:59:34Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
```

The host cluster (WGE is itself installed on a vcluster, that spawns other vclusters)

```
$ k version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.1", GitCommit:"4c9411232e10168d7b050c49a1b59f6df9d7ea4b", GitTreeState:"clean", BuildDate:"2023-04-14T13:14:41Z", GoVersion:"go1.20.3", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.16", GitCommit:"60e5135f758b6e43d0523b3277e8d34b4ab3801f", GitTreeState:"clean", BuildDate:"2023-01-18T15:54:23Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.27) and server (1.23) exceeds the supported minor version skew of +/-1
```

**Priority**
 - low
 - medium
 - high

**Severity**
 - low
 - medium
 - high
 - critical

**Describe the bug**
```
2023/05/19 14:42:10 http: panic serving 10.32.0.8:34990: runtime error: index out of range [0] with length 0
goroutine 9949 [running]:
net/http.(*conn).serve.func1()
	/usr/local/go/src/net/http/server.go:1854 +0xbf
panic({0x35a5820, 0xc00473a588})
	/usr/local/go/src/runtime/panic.go:890 +0x263
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/templates.listNodeFields({0xc0028ae800?, 0x15, 0x0?})
	/app/cmd/clusters-service/pkg/templates/text_parser.go:26 +0x6fb
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/templates.listNodeFields({0xc00438e6c0?, 0x1, 0x2?})
	/app/cmd/clusters-service/pkg/templates/text_parser.go:30 +0x3ce
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/templates.parseTextParamNames(0xc004768020?)
	/app/cmd/clusters-service/pkg/templates/text_parser.go:13 +0x45
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/templates.(*TextTemplateProcessor).ParamNames(0xc0046af5c0?, {0xc00439c400?, 0x388c25e?, 0x22?})
	/app/cmd/clusters-service/pkg/templates/processors.go:331 +0xb9
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/templates.ParseTemplateMeta({0x3f96f88, 0xc0030baf00}, {0x388c25e, 0x22})
	/app/cmd/clusters-service/pkg/templates/meta.go:35 +0x422
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/server.getProvider({0x3f96f88?, 0xc0030baf00?}, {0x388c25e?, 0xa?})
	/app/cmd/clusters-service/pkg/server/helpers.go:56 +0x45
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/server.ToTemplateResponse({0x3f96f88, 0xc0030baf00})
	/app/cmd/clusters-service/pkg/server/template_response.go:30 +0x21d
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/server.(*server).ListTemplates(0xc000752c00, {0x3f902a8, 0xc003faaba0}, 0xc003f4e410)
	/app/cmd/clusters-service/pkg/server/templates.go:109 +0xc5e
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/protos.local_request_ClustersService_ListTemplates_0({0x3f902a8, 0xc003faaba0}, {0x322d8a0?, 0x5d43aa8?}, {0x3fc2ff0, 0xc000752c00}, 0xc003e2ec00, 0x30?)
	/app/cmd/clusters-service/pkg/protos/cluster_services.pb.gw.go:65 +0x163
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/protos.RegisterClustersServiceHandlerServer.func1({0x3f7f860, 0xc0030033e0}, 0xc003e2ec00, 0xc00068fda0?)
	/app/cmd/clusters-service/pkg/protos/cluster_services.pb.gw.go:1766 +0x213
github.com/grpc-ecosystem/grpc-gateway/v2/runtime.(*ServeMux).ServeHTTP(0xc00076d580, {0x3f7f860, 0xc0030033e0}, 0xc003e2ec00)
	/go/pkg/mod/github.com/grpc-ecosystem/grpc-gateway/v2@v2.10.2/runtime/mux.go:386 +0x9e6
github.com/weaveworks/weave-gitops/pkg/server/middleware.WithLogging.func1({0x3f8de40?, 0xc001cae460}, 0xc003e2ec00)
	/go/pkg/mod/github.com/weaveworks/weave-gitops@v0.23.0/pkg/server/middleware/middleware.go:52 +0xe7
net/http.HandlerFunc.ServeHTTP(0xc003e2eb00?, {0x3f8de40?, 0xc001cae460?}, 0x5d43aa8?)
	/usr/local/go/src/net/http/server.go:2122 +0x2f
github.com/weaveworks/weave-gitops/pkg/server/auth.(*authenticatedMiddleware).ServeHTTP(0xc00012e5a0, {0x3f8de40, 0xc001cae460}, 0xc003e2eb00)
	/go/pkg/mod/github.com/weaveworks/weave-gitops@v0.23.0/pkg/server/auth/auth.go:255 +0x5ca
github.com/weaveworks/weave-gitops/pkg/server/middleware.WithProviderToken.func1({0x3f8de40, 0xc001cae460}, 0xc003e2eb00)
	/go/pkg/mod/github.com/weaveworks/weave-gitops@v0.23.0/pkg/server/middleware/middleware.go:93 +0x334
net/http.HandlerFunc.ServeHTTP(0xc112009041e94af2?, {0x3f8de40?, 0xc001cae460?}, 0x0?)
	/usr/local/go/src/net/http/server.go:2122 +0x2f
github.com/weaveworks/weave-gitops-enterprise/common/entitlement.CheckEntitlementHandler.func1({0x3f8de40, 0xc001cae460}, 0xc003e2eb00)
	/app/common/entitlement/entitlement.go:91 +0x3f9
net/http.HandlerFunc.ServeHTTP(0x3f90200?, {0x3f8de40?, 0xc001cae460?}, 0xc003f5b490?)
	/usr/local/go/src/net/http/server.go:2122 +0x2f
github.com/weaveworks/weave-gitops-enterprise/common/entitlement.EntitlementHandler.func1({0x3f8de40, 0xc001cae460}, 0xc003e2ea00)
	/app/common/entitlement/entitlement.go:57 +0x18d
net/http.HandlerFunc.ServeHTTP(0xc001a31f00?, {0x3f8de40?, 0xc001cae460?}, 0x41052a?)
	/usr/local/go/src/net/http/server.go:2122 +0x2f
net/http.(*ServeMux).ServeHTTP(0x0?, {0x3f8de40, 0xc001cae460}, 0xc003e2ea00)
	/usr/local/go/src/net/http/server.go:2500 +0x149
net/http.serverHandler.ServeHTTP({0xc003faa180?}, {0x3f8de40, 0xc001cae460}, 0xc003e2ea00)
	/usr/local/go/src/net/http/server.go:2936 +0x316
net/http.(*conn).serve(0xc003d0db90, {0x3f902a8, 0xc001661380})
	/usr/local/go/src/net/http/server.go:1995 +0x612
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3089 +0x5ed
```

**To Reproduce**
Not sure what has caused this condition. The cluster that was installed, ran out the clock and failed catastrophically. I reinstalled everything from scratch with GitOps, and WGE has been on the unattended upgrade cycle for a while.

When I visit the templates page, I get an error when I go to the create a cluster page. There is no feedback from the UI.

https://www.youtube.com/watch?v=UErP1NEECI8&t=65s

Here is the UI at that moment (no feedback).

The error comes through the clusters-service, not sure how to interpret this:

```
2023/05/19 14:42:10 http: panic serving 10.32.0.8:34990: runtime error: index out of range [0] with length 0
goroutine 9949 [running]:
net/http.(*conn).serve.func1()
	/usr/local/go/src/net/http/server.go:1854 +0xbf
panic({0x35a5820, 0xc00473a588})
	/usr/local/go/src/runtime/panic.go:890 +0x263
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/templates.listNodeFields({0xc0028ae800?, 0x15, 0x0?})
	/app/cmd/clusters-service/pkg/templates/text_parser.go:26 +0x6fb
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/templates.listNodeFields({0xc00438e6c0?, 0x1, 0x2?})
	/app/cmd/clusters-service/pkg/templates/text_parser.go:30 +0x3ce
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/templates.parseTextParamNames(0xc004768020?)
	/app/cmd/clusters-service/pkg/templates/text_parser.go:13 +0x45
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/templates.(*TextTemplateProcessor).ParamNames(0xc0046af5c0?, {0xc00439c400?, 0x388c25e?, 0x22?})
	/app/cmd/clusters-service/pkg/templates/processors.go:331 +0xb9
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/templates.ParseTemplateMeta({0x3f96f88, 0xc0030baf00}, {0x388c25e, 0x22})
	/app/cmd/clusters-service/pkg/templates/meta.go:35 +0x422
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/server.getProvider({0x3f96f88?, 0xc0030baf00?}, {0x388c25e?, 0xa?})
	/app/cmd/clusters-service/pkg/server/helpers.go:56 +0x45
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/server.ToTemplateResponse({0x3f96f88, 0xc0030baf00})
	/app/cmd/clusters-service/pkg/server/template_response.go:30 +0x21d
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/server.(*server).ListTemplates(0xc000752c00, {0x3f902a8, 0xc003faaba0}, 0xc003f4e410)
	/app/cmd/clusters-service/pkg/server/templates.go:109 +0xc5e
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/protos.local_request_ClustersService_ListTemplates_0({0x3f902a8, 0xc003faaba0}, {0x322d8a0?, 0x5d43aa8?}, {0x3fc2ff0, 0xc000752c00}, 0xc003e2ec00, 0x30?)
	/app/cmd/clusters-service/pkg/protos/cluster_services.pb.gw.go:65 +0x163
github.com/weaveworks/weave-gitops-enterprise/cmd/clusters-service/pkg/protos.RegisterClustersServiceHandlerServer.func1({0x3f7f860, 0xc0030033e0}, 0xc003e2ec00, 0xc00068fda0?)
	/app/cmd/clusters-service/pkg/protos/cluster_services.pb.gw.go:1766 +0x213
github.com/grpc-ecosystem/grpc-gateway/v2/runtime.(*ServeMux).ServeHTTP(0xc00076d580, {0x3f7f860, 0xc0030033e0}, 0xc003e2ec00)
	/go/pkg/mod/github.com/grpc-ecosystem/grpc-gateway/v2@v2.10.2/runtime/mux.go:386 +0x9e6
github.com/weaveworks/weave-gitops/pkg/server/middleware.WithLogging.func1({0x3f8de40?, 0xc001cae460}, 0xc003e2ec00)
	/go/pkg/mod/github.com/weaveworks/weave-gitops@v0.23.0/pkg/server/middleware/middleware.go:52 +0xe7
net/http.HandlerFunc.ServeHTTP(0xc003e2eb00?, {0x3f8de40?, 0xc001cae460?}, 0x5d43aa8?)
	/usr/local/go/src/net/http/server.go:2122 +0x2f
github.com/weaveworks/weave-gitops/pkg/server/auth.(*authenticatedMiddleware).ServeHTTP(0xc00012e5a0, {0x3f8de40, 0xc001cae460}, 0xc003e2eb00)
	/go/pkg/mod/github.com/weaveworks/weave-gitops@v0.23.0/pkg/server/auth/auth.go:255 +0x5ca
github.com/weaveworks/weave-gitops/pkg/server/middleware.WithProviderToken.func1({0x3f8de40, 0xc001cae460}, 0xc003e2eb00)
	/go/pkg/mod/github.com/weaveworks/weave-gitops@v0.23.0/pkg/server/middleware/middleware.go:93 +0x334
net/http.HandlerFunc.ServeHTTP(0xc112009041e94af2?, {0x3f8de40?, 0xc001cae460?}, 0x0?)
	/usr/local/go/src/net/http/server.go:2122 +0x2f
github.com/weaveworks/weave-gitops-enterprise/common/entitlement.CheckEntitlementHandler.func1({0x3f8de40, 0xc001cae460}, 0xc003e2eb00)
	/app/common/entitlement/entitlement.go:91 +0x3f9
net/http.HandlerFunc.ServeHTTP(0x3f90200?, {0x3f8de40?, 0xc001cae460?}, 0xc003f5b490?)
	/usr/local/go/src/net/http/server.go:2122 +0x2f
github.com/weaveworks/weave-gitops-enterprise/common/entitlement.EntitlementHandler.func1({0x3f8de40, 0xc001cae460}, 0xc003e2ea00)
	/app/common/entitlement/entitlement.go:57 +0x18d
net/http.HandlerFunc.ServeHTTP(0xc001a31f00?, {0x3f8de40?, 0xc001cae460?}, 0x41052a?)
	/usr/local/go/src/net/http/server.go:2122 +0x2f
net/http.(*ServeMux).ServeHTTP(0x0?, {0x3f8de40, 0xc001cae460}, 0xc003e2ea00)
	/usr/local/go/src/net/http/server.go:2500 +0x149
net/http.serverHandler.ServeHTTP({0xc003faa180?}, {0x3f8de40, 0xc001cae460}, 0xc003e2ea00)
	/usr/local/go/src/net/http/server.go:2936 +0x316
net/http.(*conn).serve(0xc003d0db90, {0x3f902a8, 0xc001661380})
	/usr/local/go/src/net/http/server.go:1995 +0x612
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3089 +0x5ed
```

**Actual behaviour**
You do not get the "create a cluster" workflow

**Expected behaviour**
This cluster has several GitopsTemplates deployed on it, and I should be able to instantiate them normally

**Cloud provider:**
 - Bare metal (kubeadm) with vcluster CAPI provider

**Additional context**

The same entitlement is installed on the cluster, I don't have any reason to expect that my entitlement should have gone expired (and I don't assume that has anything to do with the error message, even though I see the EntitlementCheckHandler popping up several times in the error – if this is what expired entitlements look like, then I guess that's a second bug)
